src/stores/game.js

```diff
@@ -165,33 +164,75 @@ export const useGameStore = defineStore('game', {
-        } else if (cardKeys.includes(key)) {
-          this[key] = createGameCard(newGame[key]) ?? null;
-        } else if (key === 'players') {
-          this.players = newGame.players?.map((player) => setPlayers(player, this.myPNum)) ?? null;
+      const updateGameMap = {
+        id: (newGame) => (this.id = newGame.id),
+        turn: (newGame) => (this.turn = newGame.turn),
+        chat: (newGame) => (this.chat = cloneDeep(newGame.chat)),
+        deck: (newGame) => (this.deck = newGame.deck.map((card) => createGameCard(card))),
+        scrap: (newGame) => (this.scrap = newGame.scrap.map((card) => createGameCard(card))),
+        log: (newGame) => (this.log = cloneDeep(newGame.log)),
+        name: (newGame) => (this.name = newGame.name),
+        p0Ready: (newGame) => (this.p0Ready = newGame.p0Ready),
+        p1Ready: (newGame) => (this.p1Ready = newGame.p1Ready),
+        passes: (newGame) => (this.passes = newGame.passes),
+        players: (newGame) => {
+          this.players = newGame.players.map((player) => setPlayers(player, this.myPNum)) ?? null;
+        },
+        spectatingUsers: (newGame) => (this.spectatingUsers = newGame.spectatingUsers),
+        twos: (newGame) => (this.twos = newGame.twos.map((card) => createGameCard(card))),
+        topCard: (newGame) => (this.topCard = createGameCard(newGame.topCard) ?? null),
+        secondCard: (newGame) => (this.secondCard = createGameCard(newGame.secondCard) ?? null),
+        oneOff: (newGame) => (this.oneOff = createGameCard(newGame.oneOff) ?? null),
+        oneOffTarget: (newGame) => (this.oneOffTarget = createGameCard(newGame.oneOffTarget) ?? null),
+        isRanked: (newGame) => (this.isRanked = newGame.isRanked),
+        currentMatch: (newGame) => (this.currentMatch = newGame.currentMatch),
+        lastEvent: (newGame) => {
+          const lastEventMap = {
+            change: (newGame) => (this.lastEventChange = newGame.lastEvent.change),
+            oneOff: (newGame) => (this.lastEventOneOffRank = newGame.lastEvent.oneOff.rank),
+            oneOffTargetType: (newGame) => (this.lastEventTargetType = newGame.lastEvent.oneOffTargetType),
+            discardedCards: (newGame) => (this.discardedCards = newGame.lastEvent.discardedCards),
+            chosenCard: (newGame) => {
+              this.cardChosenFromScrap = newGame.lastEvent.chosenCard;
+              this.playerChoosingFromScrap = newGame.lastEvent.pNum === this.myPNum;
+            },
+          };
+          Object.keys(newGame.lastEvent).forEach((lastEventKey) => lastEventMap[lastEventKey](newGame));
+        },
+      };
+      //loop through all keys sent through socket message and run corresponding function
+      Object.keys(newGame).forEach((key) => {
+        if (updateGameMap.key) {
+          updateGameMap[key](newGame);
         } else {
           this[key] = newGame[key] ?? null;
         }
       });
+
+      // Object.keys(newGame).forEach((key) => {
```

Comment:
NoI need to keep the commented out code, you can remove it. How do you feel about this new solution? Feel better or worse? Penny for your thoughts.

Reply:
I think its a great idea honestly, It makes it clearer where everything is being set where as in my code, if it wasn't a edge case it wasn't clear exactly what was being set

Reply:
Thanks for the input!!